job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -193,12 +189,7 @@ public WorkflowRun(WorkflowJob job, File dir) throws IOException {
             execution = newExecution;
             executionPromise.set(newExecution);
         } catch (Throwable x) {
-            if (listener == null) {
-                LOGGER.log(Level.WARNING, this + " failed to start", x);
-            } else {
-                x.printStackTrace(listener.error("failed to start build"));
-            }
-            setResult(Result.FAILURE);
+            finish(Result.FAILURE, x);
```

Comment:
Right, I was wondering should we do this when I was making that "zombie" build fix.